addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -379,3 +382,255 @@ def _l10n_es_edi_facturae_render_facturae(self):
+                    % Markup().join(Markup("<li>%s</li>") % log for log in logs)
+
+        invoice.message_post(body=body)
+
+        return logs
+
+    def _import_invoice_fill_lines(self, invoice, tree, ref_multiplier):
+        lines = tree.xpath('.//InvoiceLine')
+        logs = []
+        for line in lines:
+            invoice_line = invoice.invoice_line_ids.create({'move_id': invoice.id})
+
+            # ==== name ====
+            item_description = find_xml_value('.//ItemDescription', line)
+            if item_description:
+                product = self._search_product_for_import(item_description)
+                if product:
+                    invoice_line.product_id = product
+                else:
+                    logs.append(_("The product '%s' could not be found.", item_description))
+                invoice_line.name = item_description
+
+            # ==== quantity ====
+            invoice_line.quantity = find_xml_value('.//Quantity', line) or 1
+
+            # ==== price_unit ====
+            price_unit = find_xml_value('.//UnitPriceWithoutTax', line)
+            if price_unit:
+                invoice_line.price_unit = ref_multiplier * float(price_unit)
+
+            # ==== discount ====
+            discounts = line.xpath('.//DiscountRate')
+            discount_rate = 0.0
+            for discount in discounts:
+                discount_rate += float(discount.text)
+
+            charges = line.xpath('.//ChargeRate')
+            charge_rate = 0.0
+            for charge in charges:
+                charge_rate += float(charge.text)
+
+            discount_rate -= charge_rate
+            invoice_line.discount = discount_rate
+
+            # ==== tax_ids ====
+            taxes_withheld_nodes = line.xpath('.//TaxesWithheld/Tax')
+            taxes_outputs_nodes = line.xpath('.//TaxesOutputs/Tax')
+            is_purchase = invoice.move_type.startswith('in')
+            invoice_line.tax_ids = False
+            logs += self._import_fill_invoice_line_taxes(invoice_line, taxes_outputs_nodes, False, is_purchase)
+            logs += self._import_fill_invoice_line_taxes(invoice_line, taxes_withheld_nodes, True, is_purchase)
+
+        return logs
+
+    def _import_fill_invoice_line_taxes(self, invoice_line, tax_nodes, is_withheld, is_purchase):
+        logs = []
+        for tax_node in tax_nodes:
+            # Since the 'TaxRate' node isn't guaranteed to be a percentage, we can find out by
+            # applying the tax rate on the taxable base, and if it's equal to the tax amount
+            # then we can say this is a percentage, otherwise a fixed amount.
+            tax_rate = find_xml_value('.//TaxRate', tax_node)
+            taxable_base = find_xml_value('.//TaxableBase/TotalAmount', tax_node)
+            tax_amount = find_xml_value('.//TaxAmount/TotalAmount', tax_node)
+            is_fixed = False
+
+            if tax_rate and taxable_base and tax_amount:
```

Comment:
What if you have a tax rate but not the others?
We just don't do anything silently?